src/chef-server-ctl/plugins/reindex.rb

```diff
@@ -63,6 +63,27 @@ def do_reindex(orgs_to_reindex, options)
   enable_api if options[:disable_api]
 end
 
+# Disk space verification for running reindex 
+# It needs enough disk space to hold a second copy of the data plus whatever change in size the new index might need if the schema has changed.
+# Required Disk Space = (2 * size of ES data directory)
+def verify_disk_space
+  data_dir = running_config['private_chef']['elasticsearch']['data_dir']
+  if Dir.exist?(data_dir)
+    data_dir_size = Du.du(data_dir)
+    free_disk_space = Statfs.new("#{data_dir}/..").free_space
+    if (2.2* data_dir_size) < free_disk_space #The minimum space should be double the existing data size
+      puts "Free space is sufficient to start Elasticsearch reindex"
+    else
+      puts "Insufficient free space on disk to complete reindex."
+      puts "The current elasticsearch data directory contains #{data_dir_size} KB of data but only #{free_disk_space} KB is available on disk."
+      puts "The reindex process requires at least #{2*data_dir_size} KB."
+      exit 1
+    end
+  else
+    puts "Elasticserach data path does not exist, so skipping the disk space preflight validations: #{data_dir}"
+  end
```

Comment:
This should end in a `rescue` that shows only the error message/info in case of failure.  That way  if anything in `Du.du` fails it doesn't raise a stack-trace - those are noise and generally unhelpful to the operator. 

To that end, it would be better if `du` failures raised a known type (like a new exception DuFailedException) so that we could rescue it specifically here.